### PR TITLE
feat: Add Gauge Circle component and documentation

### PR DESCRIPTION
**Pull request by luis-codex**
Hello

This PR adds the GaugeCircle component, which is used to display a circular gauge with a percentage value.

[Gauge Circle _ Magic UI y 1 página más - Personal_ Microsoft_ Edge Beta 2024-05-28 14-44-00.webm](https://github.com/magicuidesign/magicui/assets/101147375/4f423e93-e458-47a6-99f6-7825d8425583)

### CODE REVIEW

File: registry/components/magicui/gauge-circle.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  min: number;
+  gaugePrimaryColor: string;
+  gaugeSecondaryColor: string;
+  className?: string;
+}
+
+/*
+  This is the css that should be added to a css file
+ */
+const css = `.gauge_circle {
```
Comment (dillionverma) [inline registry/components/magicui/gauge-circle.tsx:15]: can we just convert this to inline css variables + tailwindcss?

You already started the process below a bit but looks like we didn't finish it

File: registry/components/magicui/gauge-circle.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+.gauge_animate .gauge_content {
+    animation: gauge_fadeIn var(--transition-length) ease forwards;
+    animation-delay: var(--delay)
+}
+
+@keyframes gauge_fadeIn {
+    to {
+        opacity: 1
```
Comment (dillionverma) [inline registry/components/magicui/gauge-circle.tsx:46]: this should be added to a tailwind.config.js file and then also written in the `gauge-circle.mdx` so users know how to set it up

**Review by dillionverma — CHANGES_REQUESTED** [review]
Really appreciate that you're adding this in dude! Thanks  for the contribution

I added a few comments to ensure the code quality is good / consistent with the rest of the components. 

Let me know if the feedback makes sense!

File: registry/components/magicui/gauge-circle.tsx (hunk 1 above)
Comment (luis-codex) [reply registry/components/magicui/gauge-circle.tsx:15]: Hi @dillionverma,

Yes, we can convert the CSS to inline CSS variables and use TailwindCSS.

File: registry/components/magicui/gauge-circle.tsx (hunk 2 above)
Comment (luis-codex) [reply registry/components/magicui/gauge-circle.tsx:46]: Got it. I'll add the  to the  file and update the  documentation so users know how to set it up.

Comment (luis-codex) [reply registry/components/magicui/gauge-circle.tsx]: No need to add the @keyframes gauge_fadeIn in the tailwind.config.js file, I solved it this way.
image

**Review by dillionverma — APPROVED** [review]
LFG LGTM TYSM 🔥

**Pull request merged**